src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -306,6 +306,17 @@ void java_bytecode_convert_classt::convert(const classt &c)
     }
   }
 
+  // now do lambda method handles (bootstrap methods)
+  for(const auto &lambda_entry : c.lambda_method_handle_map)
+  {
+    const reference_typet lambda_ref = java_reference_type(
```

Comment:
Now I think about it a bit more, seems odd that this is a `reference_typet`, why not just directly store either:

 - `symbol_typet` 
 - `symbol_exprt` of type the type of the method (e.g. a `code_typet`)
 - just an `irep_idt` as the identifier 

Again interested in other opinions on this?

Reply:
The reason why I opted for a reference here is that method handles are defined as references to methods: https://docs.oracle.com/javase/8/docs/api/java/lang/invoke/MethodHandle.html
I'm happy to change it to any of the suggested types if that would make it semantically clearer or more fitting to the CBMC logic.

Reply:
Interesting - I think you're right that my suggestions aren't quite right.

However, I think what you have might be a bit of a mismatch. It seems like the _type_ of the method handle is a `java_reference_type(code_typet{params and returns})` (e.g. kind of like `Predicate<Integer>` and the _value_ is a `address_of_exprt(symbol_exprt{method_name}))` (e.g. `Integer::equals`). 

I don't think it will cause problems, but for example I would expect if I look up a `symbol_typet` in the symbol table, then what I'd get back would have `is_type` set to true, wheras in this case it will be `is_function()` that returns true. 

Not sure - thinking in comments - do you have any thoughts on this @mgudemann ?

Reply:
After an offline discussion with @thk123 the type of a method handle is now `symbol_exprt`. Any further thoughts on this choice would be appreciated.